### ISSUE TITLE
Store per profile settings in the database

### ISSUE THREAD

**Issue by sudden6** [qTox/qTox#5382]
Currently we have a per profile *.ini file as well as a sqlite database. To reduce complexity we should move everything stored in the *.ini file to the database.

The switch has the following advantages:

- less files that the user has to backup
- probably easier settings upgrades

Disadvantages:

- some work
- Need to find a good db schema that allows future upgrades and allows us to change datatypes later

@Diadlo @sphaerophoria what do you think?

**Comment by sphaerophoria** [comment]
I think well need to take a closer look at whats in the ini file. It may be good to have the settings accessible from a text editor if the user somehow hits a bug that puts them in an unusable state. This also has the benefit of the user being able to smoke program state separate from the db which has data they might care about more.

**Comment by sudden6** [comment]
I think especially the settings from the group `Friends`, `Requests` and `Circles` don't really belong into a ini style file.

I agree that it would make it harder to separate history and settings, but it should still be possible to make these different tables in the database and pretty much independent. In the current state the settings file is also not really an easy to edit file, because it contains some binary data mixed in.

**Comment by Diadlo** [comment]
I agree, that `Friend`, `Requests`, etc. should be moved out of .ini settings file. But real settings should be left as is

**Comment by sphaerophoria** [comment]
I'm okay with that.

**Comment by sudden6** [comment]
Ok, then lets move these to the db.

**Comment by sphaerophoria** [comment]
I'd like to block this on #5376 if possible

I'm touching the friend options area fairly heavily (default upgrades in #5380 followed by a few more changes on #5376) non-trivially, merge conflicts could be avoided if we postpone this a little bit.

**Comment by sudden6** [comment]
I see no problem with postponing this.
